pkg/threescale/threescale.go

```diff
@@ -284,6 +296,22 @@ func parseActionPath(path string) (string, error) {
 	return userKey, nil
 }
 
+// rpcStatusErrorHandler provides a uniform way to log and format error messages and status which should be
+// returned to the user in cases where the authorization request is rejected.
+func rpcStatusErrorHandler(userFacingErrMsg string, fn func(string) rpc.Status, err error) (rpc.Status, error) {
+	if userFacingErrMsg != "" {
+		err = fmt.Errorf("%s - %s", userFacingErrMsg, err.Error())
+	}
+
+	log.Error(err.Error())
+	return fn(err.Error()), err
+}
+
+//TODO - This should be replaced by function in istio in 1.1
```

Comment:
this comment should be expanded a bit - which function, why, etc?

Reply:
Will do